go/chat/searcher.go

```diff
@@ -0,0 +1,90 @@
+package chat
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+)
+
+const pageSize = 300
+
+type Searcher struct {
+	globals.Contextified
+	utils.DebugLabeler
+}
+
+func NewSearcher(g *globals.Context) *Searcher {
+	labeler := utils.NewDebugLabeler(g.GetLog(), "searcher", false)
+	return &Searcher{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: labeler,
+	}
+}
+
```

Comment:
I think this should be called like `SearchRegexp` and take the regexp as a parameter. That way, in case I want to compile it externally and keep re-using it I can.